app/system/controllers/Settings.php

```diff
@@ -238,4 +235,29 @@ protected function validateSettingItems($skipSession = FALSE)
 
         return $this->settingItemErrors = $settingItemErrors;
     }
+
+    protected function validateFormRequest($model)
+    {
+        $requestClass = $this->formWidget->config['request'] ?? false;
+
+        if ($requestClass === FALSE)
+            return;
+
+        if (!class_exists($requestClass))
+            throw new ApplicationException(sprintf(lang('admin::lang.form.request_class_not_found'), $requestClass));
+
+        $this->resolveFormRequest($requestClass);
+    }
+
+    protected function resolveFormRequest($requestClass)
+    {
+        app()->resolving($requestClass, function ($request, $app) {
+            if (method_exists($request, 'setController'))
+                $request->setController($this);
+
+            $request->setInputKey('Setting');
+        });
+
+        return app()->make($requestClass);
+    }
```

Comment:
@sampoyigi what do you think of this approach?

Reply:
Looks good.